### PR TITLE
Refactor WaitStrategies & add SimpleWaitStrategy and OutputWaitStrategy

### PR DESCRIPTION
**Pull request by qoomon**

### CODE REVIEW

**Comment by qoomon** [comment]
I added two wait strategies
SimpleWaitStrategiy: for common wait strategies e.g. in the Wait class static method Wait.until
OutputWaitStrategy: waits for specific output of the container

File: core/src/main/java/org/testcontainers/containers/wait/Wait.java [hunk 1]
```diff
@@ -51,4 +54,52 @@ public static HttpWaitStrategy forHttps(String path) {
         return forHttp(path)
                 .usingTls();
     }
+
+    /**
+     * Convenience method to return a WaitStrategy for a generic {@link SimpleWaitStrategy.ContainerReadyCheckFunction}.
+     * <p>
+     * <b>Example</b>
+     * <pre>{@code
+     * Wait.forCheck( "send smtp 'HELO' command", container -> {
```
Comment (dbyron0) [inline core/src/main/java/org/testcontainers/containers/wait/Wait.java]: s/forCheck/until, yes?

Comment (qoomon) [inline core/src/main/java/org/testcontainers/containers/wait/Wait.java]: oh yes

**Comment by qoomon** [comment]
Perfect if you need any help don't make mind to ask.

Great Library so.

So long
Bengt

On Sun, 3 Jul 2016, 15:42 Richard North, <notifications@github.com> wrote:

> Thank you for this, and sorry for the delay in looking at it properly. We
> seem to have some test failures, so if you don't mind I'll delay merging
> until I can resolve - probably on top of the refactoring that I'm doing for
> v1.1. I hope that's OK - of course if you can address these I'll try and
> move this up the priority list!
>
> Thanks
> Rich
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/testcontainers/testcontainers-java/pull/164#issuecomment-230154085>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADx6AoC4IVljq8YEhdQrfCOco1O63xZCks5qR7xDgaJpZM4I9G7q>
> .
>
-- 

Send from Mobile

**Comment by qoomon** [comment]
I messed it up have to fix it tonight

**Pull request closed without merging**